File: 프로그래머스/2/132265. 롤케이크 자르기/롤케이크 자르기.py
def solution(topping):
    answer = 0
    topping_l = [0]*(max(topping)+1)
    topping_r = [0]*(max(topping)+1)
    l_count = 0
    r_count = 0
    for topp in topping:
        topping_r[topp] += 1
        if topping_r[topp] == 1:
            r_count+=1
    for topp in topping:
        topping_r[topp] -= 1
        topping_l[topp] += 1
        if topping_r[topp] == 0:
            r_count-=1
        if topping_l[topp] == 1:
            l_count+=1
        if r_count == l_count:
            answer+=1
    return answer

File: 프로그래머스/2/132265. 롤케이크 자르기/롤케이크 자르기.js
function solution(topping) {
    let answer = 0;
    let r = Array(10001).fill(0) 
    let l = Array(10001).fill(0)
    let r_c = 0
    let l_c = 0
    for (let i = 0; i < topping.length; i++) {
        r[topping[i]] += 1
        if (r[topping[i]] === 1) {
           r_c += 1 
        }
    }
    for (let i = 0; i < topping.length; i++) {
        r[topping[i]] -= 1
        l[topping[i]] += 1
        if (r[topping[i]] === 0) {
            r_c -= 1
        }
        if (l[topping[i]] === 1) {
            l_c += 1
        }
        if (l_c === r_c) {
            answer += 1
        } else if (l_c > r_c) {
            return answer
        }
    }
    return answer;
}